sqlglot/dialects/spark.py

```diff
@@ -90,6 +90,8 @@ def _dateadd_sql(self: Spark.Generator, expression: exp.TsOrDsAdd | exp.Timestam
 
 
 class Spark(Spark2):
+    SUPPORTS_ORDER_BY_ALL = True
```

Comment:
Is this only available in Spark3?

Reply:
Yep, I tested Spark 2 and it didn't work; On the other hand, both Spark 3.4 and 3.5.1 had support for it

Reply:
Sounds good